tensorbay/client/dataset.py

```diff
@@ -487,6 +499,80 @@ def upload_segment(
             )
             raise
 
+    def _list_diff_segments(
+        self, *, source: Optional[Union[str, int]] = None, target: Optional[Union[str, int]] = None
+    ) -> PagingList[Dict[str, Any]]:
+        """List diffs of each segment between two versions.
+
+        Arguments:
+            source: Source version identification. Type int for draft number, type str for revision.
+                If is not given, use the current version.
+            target: Target version identification. Type int for draft number, type str for revision.
+                If is not given, use the parent commit id.
+
+        Examples:
+            >>> self._list_diff_segments(source="b382450220a64ca9b514dcef27c82d9a", target=1)
+            {
+                "segments": [
+                    {
+                        "name": "Segment1",
+                        "action": "add",
+                        "data": {
+                            "stats": {
+                                "total": 3,
+                                "additions": 3,
+                                "deletions": 0,
+                                "modifications": 0
+                            },
+                        },
+                        "sensors": {
+                            "action": "modify"
+                        }
+                    },
+                    {
+                        "name": "Segment2",
+                        "action": "add",
+                        "data": {
+                            "stats": {
+                                "total": 3,
+                                "additions": 3,
+                                "deletions": 0,
+                               "modifications": 0
+                            },
+                        },
+                        "sensors": {
+                            "action": "modify"
+                        }
+                    },
+                ]
+                ...
+                "offset": 0,
+                "recordSize": 2,
+                "totalCount": 2
+            }
+
+        Returns:
+            The PagingList of diff segments.
+
+        """
+        if source:
+            source = f"draft-{source}" if isinstance(source, int) else f"commit-{source}"
+        else:
+            if self.status.is_draft:
+                source = f"draft-{self.status.draft_number}"
+            else:
+                source = f"commit-{self.status.commit_id}"
+
+        if target:
+            target = f"draft-{target}" if isinstance(target, int) else f"commit-{target}"
+            basehead = f"{source}...{target}"
+        else:
+            basehead = source
```

Comment:
the logic of `source` + `target` -> `basehead` is repeated, try to wrap it as a function